src/pages/[marketplace]/deals/index.tsx

```diff
@@ -141,6 +142,18 @@ const Deals: NextPageWithLayout = () => {
 	const mapDeals = (deals: Deal[]) =>
 		deals
 			.slice()
+			// TODO: move this to the client
+			.filter((deal) => {
+				if (isUnderwriter) {
```

Comment:
can't this be simplified as
`isUnderwriter || deal.borrower.toString() === publicKey.toString()`

Reply:
I guess if your pass active but you're neither underwriter nor borrower it shouldn't show anything but I think it's better to block the route instead of displaying an empty table.
@MaximPiessen

Reply:
maybe for now we can do it like this and add a ticket?

Reply:
yeah I had the same thought, we dont check the activeness of the credixpass anywhere. But I feel that that should be included in the route guard when we implement that.